vcd/resource_vcd_tm_content_library_test.go

```diff
@@ -0,0 +1,93 @@
+//go:build tm || ALL || functional
+
+package vcd
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccVcdTmContentLibrary(t *testing.T) {
+	preTestChecks(t)
+
+	skipIfNotSysAdmin(t)
+	skipIfNotTm(t)
+
+	var params = StringMap{
+		"Name":                t.Name(),
+		"RegionStoragePolicy": testConfig.Tm.RegionStoragePolicy,
+		"Tags":                "tm",
+	}
+	testParamsNotEmpty(t, params)
+
+	configText1 := templateFill(testAccVcdTmContentLibraryStep1, params)
+	params["FuncName"] = t.Name() + "-step2"
+	configText2 := templateFill(testAccVcdTmContentLibraryStep2, params)
+
+	debugPrintf("#[DEBUG] CONFIGURATION step1: %s\n", configText1)
+	debugPrintf("#[DEBUG] CONFIGURATION step2: %s\n", configText2)
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	resourceName := "vcd_tm_content_library.cl"
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: configText1,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", t.Name()),
+					resource.TestCheckResourceAttr(resourceName, "description", t.Name()),
+					resource.TestCheckResourceAttr(resourceName, "storage_policy_ids.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "auto_attach", "true"), // TODO: TM: Test with false
+					resource.TestCheckResourceAttrSet(resourceName, "creation_date"),
+					resource.TestCheckResourceAttr(resourceName, "is_shared", "true"),        // TODO: TM: Test with false
+					resource.TestCheckResourceAttr(resourceName, "is_subscribed", "false"),   // TODO: TM: Test with true
+					resource.TestCheckResourceAttr(resourceName, "library_type", "PROVIDER"), // TODO: TM: Test with tenant catalog
```

Comment:
Would it be possible to add these TODO tests in this PR still?

Reply:
I don't think so, for publishing and subscribe we need the option to be able to publish a catalog, which does not exist yet (only in VCD).
For _tenant_ catalogs I'm not 100% sure we can do it now, but I'll investigate on next PRs.

Reply:
Got it. Thanks!